src/components/proofs/WorkProof.tsx

```diff
@@ -68,6 +69,8 @@ export default function () {
   }
 
   async function onGenerateProof(secret?: string) {
+    if (!secret || !checkDomainToken(secret)) return
```

Comment:
Shouldn't this never happen? UI should be blocking it

Reply:
Changed it